pkg/tasks/queue/pgqueue.go

```diff
@@ -87,27 +89,37 @@ const (
 
 	sqlInsertHeartbeat = `
                 INSERT INTO task_heartbeats(token, id, heartbeat)
-                VALUES ($1, $2, now())`
+                VALUES ($1, $2, statement_timestamp())`
 	sqlQueryHeartbeats = `
                 SELECT token
                 FROM task_heartbeats
-                WHERE age(now(), heartbeat) > $1`
+				WHERE age(statement_timestamp(), heartbeat) > $1`
 	sqlRefreshHeartbeat = `
                 UPDATE task_heartbeats
-                SET heartbeat = now()
+                SET heartbeat = statement_timestamp()
```

Comment:
for a transaction, `now()` returns the start of the transaction, so I changed this to `statement_timestamp()` which I think would be close enough.

Reply:
make sense 👍🏻